### DISCUSSION TITLE
String length validation for []string

### DISCUSSION THREAD

**Discussion by eest** [danielgtaylor/huma#678] [Q&A] [answered]
Hello!

Given an input struct containing a slice of strings, something like this:
```
input *struct {
  Body struct {
    Users []string `json:"users" minItems:"1" maxItems:"5" minLength:"10" maxLength:"63"`
  }
}
```
The idea above was to both limit the amount of elements in the slice, but also the length of each individual string in the slice. The latter does not seem to work as I expected (no errors, but a value outside the min/max length is passed through). What would be the correct way of expressing what I want?

**Comment by eest** [comment, marked as answer]
To try to answer myself: I happened to find https://github.com/danielgtaylor/huma/pull/513 and based on that came up with this:
```
func Ptr[T any](v T) *T {
       return &v
}

type userString string

func (us userString) Schema(_ huma.Registry) *huma.Schema {
       return &huma.Schema{
               Type:      "string",
               MinLength: Ptr(10),
               MaxLength: Ptr(63),
       }
}

input *struct {
  Body struct {
    Users []userString `json:"users" minItems:"1" maxItems:"5"`
  }
}
```

This seems to work as expected, so while I am still curious if this is the "right" way, at least I can make something work :).

**Reply by danielgtaylor** [reply]
Yes I think this is correct, as you can't easily define the sub-schema inline via field tags. Alternatively you could use a schema transformer but it's essentially the same thing.
